sql/moz-fx-data-shared-prod/firefox_desktop/newtab_visits_daily/view.sql

```diff
@@ -1,31 +1,57 @@
 CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.firefox_desktop.newtab_visits_daily`
 AS
+WITH src AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_derived.newtab_visits_daily_v2`
+),
+enriched AS (
+  SELECT
+    src.*,
+    -- Precompute once for reuse
+    mozfun.newtab.determine_grid_layout_v1(is_section, app_version, experiments) AS layout_type,
+    (
+      is_search_issued
+      OR is_content_interaction
+      OR is_topsite_interaction
+      OR is_widget_interaction
+      OR is_wallpaper_interaction
+      OR is_other_interaction
+    ) AS is_any_interaction,
+    (
+      is_content_interaction
+      OR is_topsite_interaction
+      OR is_widget_interaction
+      OR is_wallpaper_interaction
+      OR is_other_interaction
+    ) AS is_nonsearch_interaction,
+    (
+      is_content_interaction
+      AND NOT is_sponsored_content_interaction
+    ) AS is_organic_content_interaction,
+    (
+      is_topsite_interaction
+      AND NOT is_sponsored_topsite_interaction
+    ) AS is_organic_topsite_interaction,
+  FROM
+    src
+)
 SELECT
   'Firefox Desktop' AS app_name,
-  (
-    is_search_issued
-    OR is_content_interaction
-    OR is_topsite_interaction
-    OR is_widget_interaction
-    OR is_wallpaper_interaction
-    OR is_other_interaction
-  ) AS is_any_interaction,
-  (
-    is_content_interaction
-    OR is_topsite_interaction
-    OR is_widget_interaction
-    OR is_wallpaper_interaction
-    OR is_other_interaction
-  ) AS is_nonsearch_interaction,
-  (
-    is_content_interaction
-    AND NOT is_sponsored_content_interaction
-  ) AS is_organic_content_interaction,
-  (
-    is_topsite_interaction
-    AND NOT is_sponsored_topsite_interaction
-  ) AS is_organic_topsite_interaction,
-  *
+  is_any_interaction,
+  is_nonsearch_interaction,
+  is_organic_content_interaction,
+  is_organic_topsite_interaction,
+  layout_type,
+  mozfun.newtab.determine_tiles_per_row_v1(layout_type, newtab_window_inner_width) AS tiles_per_row,
```

Comment:
is tiles per row the right paradigm now that we're rolling sections out to everyone and we have different-sized cards?

Reply:
@ilanasegall Are you saying that there is a possibility for a section layout with a specific window width to have different tile row sizes within the same window?

If so, then it perhaps does not make sense to do this computation